build/Cache.js

```diff
@@ -33,64 +32,88 @@ export class ImageCache {
       })
     }
 
+    static constructPolicy(data, type){
+      if (type === 'lru'){
+        return new LRUPolicy(data);
+      }
+    }
+
+    static readQueueAndResolve(file) {
+      ImageCache.instance.loading = true;
+      ImageCache.readQueue(file)
+      .then((data) => {
+        ImageCache.instance.policy = ImageCache.constructPolicy(data, 'lru');
+        ImageCache.instance.loading = false;
+
+        //process and remove requests
+        const requests = ImageCache.instance.requestQueue;
+        requests.reverse();
+        for (let idx = requests.length-1; idx >= 0; idx-- ){
+          requests[idx](ImageCache.instance);
+          requests.splice(idx, 1);
+        }
+      })
+      .catch((err) => console.log('Error reading queue and resolvings- ', err))
+    }
+
     static save(file, queueData) {
       return new Promise((resolve, reject) => {
         const data = JSON.stringify(queueData)
         RNFetchBlob.fs.writeFile(QUEUE_DIR + file, data, 'utf8')
-        .then((result) => (process.env.NODE_ENV === "TEST") ? console.log("") : console.log('Queue successfully saved!') )
+        .then((result) => {
+          if (process.env.NODE_ENV != "TEST") {
+            console.log('Queue successfully saved!');
+          }
+          resolve();
+        })
         .catch((err) => reject(err))
       })
     }
 
-    prepareMem(data){
-      let queue = new PriorityQueue();
-      const map = {};
-      data.forEach((d) => {
-        let n = new Node(d)
-        queue.enqueue(n)
-        map[d] = n;
-      })
-      return {queue, map}
-    }
-
-    constructor(data) {
+    constructor() {
         this.cache = {};
```

Comment:
The LRUPolicy is great. Let's go one step further. 😄 

Move cache inside lru and then name it LRUCache. So LRUCache will behave as a dictionary too. Will have
- getValue(key)
- AddPair(key, value)
- size()

And internally every time you call getValue(key) or containKey(key) then you update the priority of that key.